### PR TITLE
Upgrade 3.9.6 -> 3.9.7

### PR DESCRIPTION
**Pull request by iipeace**
Signed-off-by: Peace Lee <iipeace5@gmail.com>

### CODE REVIEW

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 1]
```diff
@@ -6,17 +6,17 @@ AUTHOR = "Peace Lee <ipeace5@gmail.com>"
...
 SRC_URI = "git://github.com/iipeace/${BPN}"
 #SRCREV = "${AUTOREV}"
-SRCREV = "fef25c41efb9bde0614ea477d0b90bd9565ae0b4"
+SRCREV = "bcf107b73dcd3613cfd8f3f68402547ba8f15701"
 
 S = "${WORKDIR}/git"
 R = "${RECIPE_SYSROOT}"
 
-inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python2", "distutils", "", d)}
+inherit distutils
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: we cant depend on meta-python2, so if it still needs distutils, then please leave this check in. We want to blacklist it unless meta-python2 is included.

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 2]
```diff
@@ -33,7 +33,3 @@ do_install() {
 
 RDEPENDS_${PN} = "python-ctypes python-shell \
                   python-json python-subprocess"
-python() {
-    if 'meta-python2' not in d.getVar('BBFILE_COLLECTIONS').split():
-        raise bb.parse.SkipRecipe('Requires meta-python2 to be present.')
-}
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: does it work with python3 now ?

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb (hunk 1 above)
Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: yes, I need meta-python2 yet...

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb (hunk 2 above)
Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: yes, it works with python3 now :)

**Comment by iipeace** [comment]
@kraj 
please review again :)

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 3]
```diff
@@ -6,17 +6,17 @@ AUTHOR = "Peace Lee <ipeace5@gmail.com>"
...
 SRC_URI = "git://github.com/iipeace/${BPN}"
 #SRCREV = "${AUTOREV}"
-SRCREV = "fef25c41efb9bde0614ea477d0b90bd9565ae0b4"
+SRCREV = "52aeb6f4c536865305afa51f37166bfe46fa460d"
 
 S = "${WORKDIR}/git"
 R = "${RECIPE_SYSROOT}"
 
-inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python2", "distutils", "", d)}
+inherit distutils
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: this should be `inherit distutils3` for py3

**Comment by kraj** [comment]
also commit msg should be
```
recipename: Short summary

Long summary
Sign-off-by
```

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb (hunk 3 above)
Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: yes, I fixed it.

**Comment by iipeace** [comment]
@kraj 
please review again :)

**Comment by kraj** [comment]
thanks in master-next now

**Comment by kraj** [comment]
Fails to build without meta-python2 see [1]

[1] https://jenkins.ccp.xcal.tv/jenkins/view/YP/job/Yocto-world-arm/448/console

**Comment by iipeace** [comment]
then should I add below code?

inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python2", "distutils", "", d)}inherit distutils

python() { if 'meta-python2' not in d.getVar('BBFILE_COLLECTIONS').split(): raise bb.parse.SkipRecipe('Requires meta-python2 to be present.')}

**Comment by kraj** [comment]
perhaps in the code replace python with python3 explicitly

**Comment by iipeace** [comment]
@kraj,
how can I access to https://jenkins.ccp.xcal.tv/jenkins/view/YP/job/Yocto-world-arm/448/console ??
I can't see any register button in the page.

**Comment by kraj** [comment]
Thats not public repo and the logs for that build are gone. I should have pointed at errors.yp.org location. Let me see if I can cherry-pick it again and reporduce the error.

**Comment by kraj** [comment]
This patch is still wrong, it should still check for meta-python2 presence as such it will fail meta-oe build when meta-py2 is not present, we do not have dependency on meta-py2

**Comment by iipeace** [comment]
okay, then could you share error logs?

**Comment by iipeace** [comment]
@kraj 
could you review again?

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 4]
```diff
@@ -6,21 +6,27 @@ AUTHOR = "Peace Lee <ipeace5@gmail.com>"
...
 #SRCREV = "${AUTOREV}"
-SRCREV = "fef25c41efb9bde0614ea477d0b90bd9565ae0b4"
+SRCREV = "948f9fbccddf0225ae4da32f5828d64c4319c8cf"
 
 S = "${WORKDIR}/git"
 R = "${RECIPE_SYSROOT}"
 
-inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python2", "distutils", "", d)}
+#inherit distutils
+inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python3", "distutils", "", d)}
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb:20]: this should be `meta-python` not `meta-python3` see https://github.com/openembedded/meta-openembedded/blob/master/meta-python/conf/layer.conf#L7

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 5]
```diff
@@ -6,21 +6,27 @@ AUTHOR = "Peace Lee <ipeace5@gmail.com>"
...
 
 GUIDER_OBJ = "guider.pyc"
 GUIDER_SCRIPT = "guider"
 
+python() {
+    if 'meta-python3' not in d.getVar('BBFILE_COLLECTIONS').split():
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb:26]: same here it should be `meta-python` according to https://github.com/openembedded/meta-openembedded/blob/master/meta-python/conf/layer.conf#L7

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb (hunk 4 above)
Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: fixed

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb (hunk 5 above)
Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: fixed

**Comment by iipeace** [comment]
I changed my PR.

File: meta-oe/recipes-devtools/guider/guider_3.9.7.bb [hunk 6]
```diff
@@ -6,21 +6,27 @@ AUTHOR = "Peace Lee <ipeace5@gmail.com>"
...
 #SRCREV = "${AUTOREV}"
-SRCREV = "fef25c41efb9bde0614ea477d0b90bd9565ae0b4"
+SRCREV = "948f9fbccddf0225ae4da32f5828d64c4319c8cf"
 
 S = "${WORKDIR}/git"
 R = "${RECIPE_SYSROOT}"
 
-inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python2", "distutils", "", d)}
+#inherit distutils
+inherit ${@bb.utils.contains("BBFILE_COLLECTIONS", "meta-python", "distutils", "", d)}
```
Comment (kraj) [inline meta-oe/recipes-devtools/guider/guider_3.9.7.bb:20]: meta-python does not provide distutils but it provides distutils3, have you tested this patch without meta-python2 ?

Comment (iipeace) [reply meta-oe/recipes-devtools/guider/guider_3.9.7.bb]: I changed the recipe again and tested it.

**Comment by kraj** [comment]
I have merged it now thanks for persisting :)

**Comment by shr-project** [comment]
@iipeace This new version looks much better without the dependency on meta-python2, but rewritting the git history of master branch broke guider recipe e.g. in zeus branch of meta-openembedded where it now fails to fetch:

```
ERROR: guider-3.9.4+gitAUTOINC+b433f38056-r0 do_fetch: Fetcher failure: Unable to find revision b433f3805674ef4588c1a161986c74eeac6a48e7 in branch master even from upstream
```

before re-write
```
guider$ git branch -a --contains b433f3805674ef4588c1a161986c74eeac6a48e7
* master
  redundant
```

after:
```
guider$ git remote update
Fetching origin
remote: Enumerating objects: 423, done.
remote: Counting objects: 100% (423/423), done.
remote: Compressing objects: 100% (31/31), done.
remote: Total 5064 (delta 405), reused 409 (delta 391), pack-reused 4641
Receiving objects: 100% (5064/5064), 4.14 MiB | 5.98 MiB/s, done.
Resolving deltas: 100% (2500/2500), completed with 23 local objects.
From git://github.com/iipeace/guider
 + dc4f3ae...ab803ac master             -> master  (forced update)
 + bca0ee3...197d588 refs/pull/207/head -> refs/pull/207/head  (forced update)
 * [new ref]         refs/pull/209/head -> refs/pull/209/head
 * [new ref]         refs/pull/210/head -> refs/pull/210/head
 * [new ref]         refs/pull/214/head -> refs/pull/214/head
 * [new ref]         refs/pull/215/head -> refs/pull/215/head
 * [new ref]         refs/pull/217/head -> refs/pull/217/head
 * [new ref]         refs/pull/218/head -> refs/pull/218/head
 * [new ref]         refs/pull/220/head -> refs/pull/220/head
 * [new ref]         refs/pull/222/head -> refs/pull/222/head
 * [new ref]         refs/pull/223/head -> refs/pull/223/head
 * [new ref]         refs/pull/224/head -> refs/pull/224/head
 t [tag update]      versions/3.8.0     -> versions/3.8.0
 t [tag update]      versions/3.9.6     -> versions/3.9.6
 t [tag update]      versions/3.9.7     -> versions/3.9.7
guider$ git branch -a --contains b433f3805674ef4588c1a161986c74eeac6a48e7
  redundant
```

Please restore master branch as it was to fix released versions of meta-oe and implement new changes on top of that - and you really shouldn't be moving the tags..

**Comment by iipeace** [comment]
@shr-project,
It was my mistake to remove all useless files from git history by filter-branch command...;(
how can I recover tag commits?

**Comment by shr-project** [comment]
@iipeace 

I've created fork of your repo at:
https://github.com/shr-project/guider
and pushed the older version of master and tags I had in my local downloads directory from OE
```
To github.com:shr-project/guider.git
 + d3ad332...dc4f3ae master -> master (forced update)
 + f505d4c...a4220da versions/3.8.0 -> versions/3.8.0 (forced update)
 + 006aa86...29eb4c2 versions/3.9.6 -> versions/3.9.6 (forced update)
 + 993014a...0786c98 versions/3.9.7 -> versions/3.9.7 (forced update)
```

It doesn't have to be the latest version of master you had before re-writting the history - so just rebase whatever you have on top of this older revision - you should be also able to find your own old history in git reflog (which should show where you did the filter-branch), that would be more accurate and really the latest before re-write. It should be there, but depends on how long expire interval you're using or if you called git expire since the re-write.

For tags you can use something like:
```
git remote add shr git@github.com:shr-project/guider.git
git fetch --tags shr
# this should show that these 3 tags were updated
git push -f origin versions/3.8.0   versions/3.9.6 versions/3.9.7 
# assuming your origin is your github repo
```

**Comment by iipeace** [comment]
@shr-project,
I rebased older branch you shared and "show-ref --tags" command shows as a below.
'''
694c2bf59d1ab199dccd5673c2c4c5777b73cdc5 refs/tags/versions/2.0.0               
fb224a932cc3ae0fe734189ba8f3e5953fcaad66 refs/tags/versions/3.0.0               
eb0388935ee262b4408f623ab73907a4cc827b06 refs/tags/versions/3.7.0               
e38a93ef34e3faf9ca27c05132c80b85861ce06f refs/tags/versions/3.7.1               
f505d4cee736afdadd9c0f53fc3f0d464ae2a915 refs/tags/versions/3.8.0               
e76d3b2424294892815031bb46237d3aa43c8ef7 refs/tags/versions/3.8.7               
fb0eb4a379936a70d46e38450e114d64309e8c57 refs/tags/versions/3.8.9               
73ac11ba939a45056664ed9e1468cd312131df03 refs/tags/versions/3.9.0               
2c1035fea192de2d609e91bb34bc0eb7f22a4ede refs/tags/versions/3.9.1               
3c9755dc4b6f295552b1569b72cc2dae5ca4d5d5 refs/tags/versions/3.9.3               
bd2997de13685c6fff2b6852c317a94bc9b7d5b7 refs/tags/versions/3.9.4               
787dc60443f5f44085c09a1a9200d663ffe11ea8 refs/tags/versions/3.9.5               
006aa86cba4b0ec427702845d1cf1976a2c30498 refs/tags/versions/3.9.6               
993014a044f64d54bd0d181ee6ac04d689352e36 refs/tags/versions/3.9.7
'''

is this right?
then I am going to push them to upstream.

**Comment by shr-project** [comment]
These 3.8.0, 3.9.6, 3.9.7 tags look like what is currently on your github, not the old versions matching with commits in old master branch. Did you forget to run "git fetch --tags shr"?

You can easily check with
git branch -a --contains refs/tags/versions/3.9.7
should show the restored master branch - and similarly for 3.8.0, 3.9.6 tags and SRCREVs used in builds.

From meta-oe perspective it these 3 SRCREVs b433f3805674ef4588c1a161986c74eeac6a48e7 (zeus and warrior)  fef25c41efb9bde0614ea477d0b90bd9565ae0b4 (dunfell) and c87269d6d4669d12c99a2a7b9b07c39a5dd24ac6 (master). The first 2 exist only in the "original" master. The c87269d6d4669d12c99a2a7b9b07c39a5dd24ac6 currently in master only in the re-written master, so the recipe in meta-oe will need and SRCREV update again once you push the rebased branch.

**Comment by iipeace** [comment]
@shr-project,
I forgot to git fetch --tags :)
this is about that.

'''
From https://github.com/shr-project/guider                                      
 t [tag update]      versions/3.8.0 -> versions/3.8.0                           
 t [tag update]      versions/3.9.6 -> versions/3.9.6                           
 t [tag update]      versions/3.9.7 -> versions/3.9.7                           
peacelee@osdc:~/work/guider$ git show-ref --tags                                
694c2bf59d1ab199dccd5673c2c4c5777b73cdc5 refs/tags/versions/2.0.0               
fb224a932cc3ae0fe734189ba8f3e5953fcaad66 refs/tags/versions/3.0.0               
eb0388935ee262b4408f623ab73907a4cc827b06 refs/tags/versions/3.7.0               
e38a93ef34e3faf9ca27c05132c80b85861ce06f refs/tags/versions/3.7.1               
a4220da234c36506f7c5b69542b6a49b6d1e5796 refs/tags/versions/3.8.0               
e76d3b2424294892815031bb46237d3aa43c8ef7 refs/tags/versions/3.8.7               
fb0eb4a379936a70d46e38450e114d64309e8c57 refs/tags/versions/3.8.9               
73ac11ba939a45056664ed9e1468cd312131df03 refs/tags/versions/3.9.0               
2c1035fea192de2d609e91bb34bc0eb7f22a4ede refs/tags/versions/3.9.1               
3c9755dc4b6f295552b1569b72cc2dae5ca4d5d5 refs/tags/versions/3.9.3               
bd2997de13685c6fff2b6852c317a94bc9b7d5b7 refs/tags/versions/3.9.4               
787dc60443f5f44085c09a1a9200d663ffe11ea8 refs/tags/versions/3.9.5               
29eb4c26be50281b7d26de13dfa04d78a2651f4d refs/tags/versions/3.9.6               
0786c98b08cda833ba9e388214ba9a04daf1c233 refs/tags/versions/3.9.7
'''

**Comment by shr-project** [comment]
These tags match with what I wrote 3 comments above (https://github.com/openembedded/meta-openembedded/pull/205#issuecomment-643780428), so it's better because they match with the old ones. But it doesn't show anything about your new master branch.

**Comment by iipeace** [comment]
@shr-project,
so should I push them to upstream master?
I will do that if it can solve build problems.

**Comment by shr-project** [comment]
Yes, if you did it correctly (use "git branch --contains" to confirm as explained before) it will fix the build issues in warrior, zeus, dunfell branches. For master you will probably need to submit another SRCREV update to use corresponding SRCREV in the new branch.

**Comment by kraj** [comment]
Please push this as new pull request

**Comment by iipeace** [comment]
@shr-project,
please apply for changes.

@kraj,
I will push this as new PR.

**Comment by shr-project** [comment]
> please apply for changes.

What?

**Pull request closed without merging**